tests/test_shelf.py

```diff
@@ -109,7 +109,7 @@ def test_find_column_from_join(self):
 
         col2 = find_column(j, "census_state")
         assert isinstance(col, (ColumnElement, InstrumentedAttribute))
-        assert col == col2
+        assert col == col2  # jingwei look into this
```

Comment:
I think `__eq__` of them are overridden, so this is generating a sqlalchemy `BinaryExpression` object instead of checking their equality, which will always evaluate to `True`..
 ```python
print(type(col == col2), col == col2)
```
gives
`tests/test_shelf.py::TestFindColumn::test_find_column_from_join <class 'sqlalchemy.sql.elements.BinaryExpression'> census.state = census.state`